pkg/controller/projectreference/projectreference_controller.go

```diff
@@ -72,34 +88,176 @@ func (r *ReconcileProjectReference) Reconcile(request reconcile.Request) (reconc
+	}
+
+	// Make projectReference  be processed based on state of ProjectClaim and Project Reference
+	switch {
+	// If we are in a creating state break from the loop and conitnue to process CR
+	case projectReference.Status.State == gcpv1alpha1.ProjectReferenceStatusCreating:
+		break
+	// If projectReference and projectClaim are both ready there is nothing to do
+	case projectReference.Status.State == gcpv1alpha1.ProjectReferenceStatusReady && adapter.projectClaim.Status.State == gcpv1alpha1.ClaimStatusReady:
+		reqLogger.Info("ProjectReference and ProjectClaim CR are in READY state nothing to process.")
+		return r.doNotRequeue()
+	case projectReference.Status.State == gcpv1alpha1.ProjectReferenceStatusReady:
+		//Project Ready update matchingClaim to ready
+		adapter.projectClaim.Status.State = gcpv1alpha1.ClaimStatusReady
+		// Since conditions as of now are not inititated we need to set an empty one here
+		// This will need to removed and checked when we actually start to use conditions
+		adapter.projectClaim.Status.Conditions = []gcpv1alpha1.ProjectClaimCondition{}
+		err := r.client.Status().Update(context.TODO(), adapter.projectClaim)
+		if err != nil {
+			reqLogger.Error(err, "Error updating ProjectClaim Status")
+			return r.requeueOnErr(err)
 		}
+		return r.doNotRequeue()
 	}
-	return instance, err
+
+	// only make changes to ProjectReference if ProjelctClaim is pending
+	// if adapter.projectClaim.Status.State != gcpv1alpha1.ClaimStatusPending {
+	// 	return r.requeueAfter(5 * time.Second)
+	// }
+
+	// make sure we meet mimimum requiresments to process request and set its state to creating or error if its not supported
+	if projectReference.Status.State == "" {
+		reqLogger.Info("Checking Requirements")
+		err := adapter.checkRequirements()
+		if err != nil {
+			// TODO: add condition here SupportedRegion = false to give more information on the error state
+			reqLogger.Error(err, "Region not supported")
+			projectReference.Status.State = gcpv1alpha1.ProjectReferenceStatusError
+			err := r.client.Status().Update(context.TODO(), projectReference)
+			if err != nil {
+				reqLogger.Error(err, "Error updating ProjectReference Status")
+				return r.requeueOnErr(err)
+			}
+			return r.doNotRequeue()
+		}
+
+		reqLogger.Info(fmt.Sprintf("Setting ProjectReferenceStatus %s", gcpv1alpha1.ProjectReferenceStatusCreating))
+		// passed requirementes check set to creating
+		projectReference.Status.State = gcpv1alpha1.ProjectReferenceStatusCreating
+		err = r.client.Status().Update(context.TODO(), projectReference)
+		if err != nil {
+			reqLogger.Error(err, "Error updating ProjectReference Status")
+			return r.requeueOnErr(err)
+		}
+	}
+
+	if projectReference.Spec.GCPProjectID == "" {
+		reqLogger.Info("Creating ProjectID in ProjectReference CR")
+		err := adapter.updateProjectID()
+		if err != nil {
+			reqLogger.Error(err, "Could not update ProjectID in Project Reference CR")
+			return r.requeueOnErr(err)
+		}
+		return r.requeue()
```

Comment:
```suggestion
	return r.requeueOnErr(err)
```
Shouldn't this be just requeueOnErr? Since updateProjectID will change the ProjectReference, it will be reconciled again anyhow.